Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/GatewayService.java

```diff
@@ -125,18 +142,29 @@ public boolean discoverPatient(final PatientSearchBean patientQuerySearch) {
 
         try {
             // Call the entity/gateway Patient Discovery service
-            final PatientSearchResults patientDiscoveryResults = patientService.queryPatient(patientBean);
+            final PatientSearchResults patientDiscoveryResults = patientService.queryPatient(patientBean, assertion);
             LOG.debug("Patient Discovery call successful. Total number of patients found: {}",
                 patientDiscoveryResults.getPatientList().size());
 
             // Return false if no patient found
             if (patientDiscoveryResults.getPatientList().isEmpty()) {
                 return false;
             }
+            
+            localCorrelation = correlationService.getLocalPatient(patientDiscoveryResults);
```

Comment:
I feel like this bit of logic you added here could be implemented into the service itself. something like "retrieveOrCreateCorrelation" which will ensure that we will return a local correlation if we find one in the database, and if not, one will be created.